sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_requests.py

```diff
@@ -7,6 +7,7 @@
 import hashlib
 import base64
 import hmac
+from yarl import URL
```

Comment:
Do we need to add yarl into setup.py?

Reply:
@mccoyp and I have talked about it, we tried to install the package in a fresh environment but didn't get `yarl` installed. But since the URL usage is from `azure-storage-blob`, they don't have `yarl` in their setup.py. They may missed it, but I'm not sure.

Reply:
Yeah, I honestly think that Storage is missing a dependency there and that we may need to require `yarl`. The fact that the package doesn't get installed when `pip install`ing `azure-storage-blob` (or `azure-appconfiguration`) means that we would otherwise expect an error when loading this module.

The other option would be to `try/except` the import, which I just noticed [`azure-data-tables` currently does](https://github.com/Azure/azure-sdk-for-python/blob/8d109cafe03ecad368ba4f99b0d011de78df2eb1/sdk/tables/azure-data-tables/azure/data/tables/_authentication.py#L22). They use `yarl` in a very similar situation, but apparently for a case where `AsyncHttpTransport` is the issue instead: https://github.com/Azure/azure-sdk-for-python/blob/8d109cafe03ecad368ba4f99b0d011de78df2eb1/sdk/tables/azure-data-tables/azure/data/tables/_authentication.py#L107-L126

Reply:
And it also interests me that why our tests did not fail?

Reply:
I think `yarl` must be a chained dependency from one of our dev requirements, which is why test environments always have it installed. When I tried installing the package by itself, and then `pip install -r dev_requirements.txt` in a clean environment, I only got `yarl` after the second command

Reply:
And `azure-storage-blob` has the same try/except import for URL, we may need to follow that: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py#L17-L20

Reply:
I somehow completely missed that they imported with a `try/except` as well lol thank you for pointing that out! That would be safe to do then, in the sense that App Config wouldn't be the first package to do it. But I am curious about the case where `yarl` isn't installed but `aiohttp` is -- is that even possible?

Reply:
I am afraid it may not be correct.

If we need yarl, we should add it into setup.py (same for storage).

Reply:
Update: `yarl` is a dependency of `aiohttp`, so we can assume they come as a pair. Make sense! In that case, we should do a `try/except`

Reply:
```
C:\venvs>310_clean\Scripts\activate

(310_clean) C:\venvs>pip list
Package    Version
---------- -------
pip        22.2.2
setuptools 65.3.0
wheel      0.37.1

(310_clean) C:\venvs>pip install aiohttp
Collecting aiohttp
  Using cached aiohttp-3.8.3-cp310-cp310-win_amd64.whl (319 kB)
Collecting attrs>=17.3.0
  Using cached attrs-22.1.0-py2.py3-none-any.whl (58 kB)
Collecting async-timeout<5.0,>=4.0.0a3
  Using cached async_timeout-4.0.2-py3-none-any.whl (5.8 kB)
Collecting aiosignal>=1.1.2
  Using cached aiosignal-1.2.0-py3-none-any.whl (8.2 kB)
Collecting charset-normalizer<3.0,>=2.0
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting frozenlist>=1.1.1
  Using cached frozenlist-1.3.1-cp310-cp310-win_amd64.whl (33 kB)
Collecting multidict<7.0,>=4.5
  Using cached multidict-6.0.2-cp310-cp310-win_amd64.whl (27 kB)
Collecting yarl<2.0,>=1.0
  Using cached yarl-1.8.1-cp310-cp310-win_amd64.whl (55 kB)
Collecting idna>=2.0
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Installing collected packages: multidict, idna, frozenlist, charset-normalizer, attrs, async-timeout, yarl, aiosignal, aiohttp
Successfully installed aiohttp-3.8.3 aiosignal-1.2.0 async-timeout-4.0.2 attrs-22.1.0 charset-normalizer-2.1.1 frozenlist-1.3.1 idna-3.4 multidict-6.0.2 yarl-1.8.1

(310_clean) C:\venvs>pip show yarl
Name: yarl
Version: 1.8.1
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: c:\venvs\310_clean\lib\site-packages
Requires: idna, multidict
Required-by: aiohttp
```

Reply:
Sounds good.

Then let's use

try:
import aiohttp
import yarl
except: